common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -120,6 +120,9 @@ public static float getFloat(Object object, long offset) {
   }
 
   public static void putFloat(Object object, long offset, float value) {
+    if(value == -0.0f) {
```

Comment:
I'm fine to put this trick here, shall we also move the IsNaN logic to here as well?